src/ctap/data_formats.rs

```diff
@@ -1322,7 +1370,84 @@ mod test {
     }
 
     #[test]
-    fn test_from_into_cose_key() {
+    fn test_from_into_cose_key_cbor() {
+        for algorithm in &[ECDH_ALGORITHM, ES256_ALGORITHM] {
+            let cbor_value = cbor_map! {
+                1 => EC2_KEY_TYPE,
+                3 => algorithm,
+                -1 => P_256_CURVE,
+                -2 => [0u8; 32],
+                -3 => [0u8; 32],
+            };
+            let cose_key = CoseKey::try_from(cbor_value.clone()).unwrap();
+            let created_cbor_value = cbor::Value::from(cose_key);
+            assert_eq!(created_cbor_value, cbor_value);
+        }
+
+        let cbor_value = cbor_map! {
```

Comment:
Nit: It's usually better to have many small tests than one large test, so that:
- if the first assertion fails, it doesn't block the rest of the test cases from running,
- it's clear at first glance what failed and what succeeded, allowing easier debugging and bug fixing if some test failed.

Reply:
Sounds good, split!